lib/ecto_materialized_path/path.ex

```diff
@@ -7,7 +7,7 @@ defmodule EctoMaterializedPath.Path do
   """
 
   def cast(list) when is_list(list) do
-    path_is_correct? = Enum.all?(list, fn(path_id) -> is_integer(path_id) end)
+    path_is_correct? = Enum.all?(list, fn(path_id) -> (is_integer(path_id) || is_binary(path_id)) end)
```

Comment:
Move condition check to the new line, it'll be easier to read. Plus, make it looks better.

Like `is_regular_id?(path_id) || is_uuid?(path_id)` with 2 private methods defined. Use internal Ecto check for uuid, please.